packages/cli/src/index.ts

```diff
@@ -76,6 +77,19 @@ const reportMetrics = async (req: http.IncomingMessage, res: http.ServerResponse
   res.end(await register.metrics())
 }
 
+const hasBootnodeENRPrefix = (enr) => {
+  if (enr && enr.startsWith('enr:-')) {
```

Comment:
do all the bootnodes (i.e. `bootnode` CLI argument value and all `bootnodeList` CLI argument values) ENR prefixes start with `enr:-`? they do according to the example shown here https://eips.ethereum.org/EIPS/eip-778

but why doesn't the `enr()` validator here https://github.com/ethereumjs/ultralight/blob/master/packages/cli/src/rpc/validators.ts#L227 only check that it starts with `enr` instead of `enr:-`? shouldn't that be updated too for consistency?

Reply:
Yes, bootnode ENRs should all start with `enr:-`.

Feel free to update the `enr()` validator for consistency and add to this PR.

Reply:
i did a benchmark comparing `startsWith` and `substr` and found that `startsWith` is about 4-6x faster

```js
function v1(enr) {
  if (enr.substr(0, 3) !== 'enr') {
    return true;
  }
  return false;
}

function v2(enr) {
  if (enr && enr.startsWith('enr:-')) {
    return true
  }
  return false
}

const enr = "enr:-IS4QFV_wTNknw7qiCGAbHf6LxB-xPQCktyrCEZX-b-7PikMOIKkBg-frHRBkfwhI3XaYo_T-HxBYmOOQGNwThkBBHYDgmlkgnY0gmlwhKRc9_OJc2VjcDI1NmsxoQKHPt5CQ0D66ueTtSUqwGjfhscU_LiwS28QvJ0GgJFd-YN1ZHCCE4k"

function runPerformance(...arguments) {
    for (var i=0; i < arguments.length; i++) {
        console.time(arguments[i].name);
        arguments[i](enr);
        console.timeEnd(arguments[i].name);
    }
}

runPerformance(v1, v2);
```

output
```
v1: 0.04ms
v2: 0.009ms
```

Reply:
> bootnode ENRs should all start with enr:-.

i understand that they should all start with `enr:`, since it's mentioned in the spec https://eips.ethereum.org/EIPS/eip-778, but i still don't understand why they all have the `-` immediately after `enr:` (i.e. `enr:-`), it seem like the `-` is padding, but i don't understand why they use it at the start and not the end, and why they use `-` instead of `=`. i read that in base64 they use `=` for padding the last character and also sometimes also the second last character.

in this enr library https://github.com/sigp/enr/blob/master/src/lib.rs#L422 they show how they construct the ENR, and they say they construct a _"URL-safe base64 encoded "text" version of the ENR prefixed by "enr:"_

Reply:
i was also thinking the `isValidEnr` function that we'll use in util.ts should also verify the signature of the ENR, for example this enr library that's written in Rust allows you to do that here https://github.com/sigp/enr/blob/master/src/lib.rs#L411, but i'm not aware of a JavaScript/TypeScript library that does the same

Reply:
i tried checking it with the following with an example bootnode ENR but it doesn't pass the `isValidBase64Length` test
```
function isString(str) {
  return str && typeof str === 'string'
}

function hasValidBase64Chars(str) {
  const validBase64Chars = /[^A-Z0-9+\/=]/i;
  if (str && validBase64Chars.test(str)) {
    return true;
  }
  return false;
}

function isValidBase64Length(str) {
  if (str && str.length && str.length % 4 == 0) {
    return true;
  }
  return false
}

function hasValidEnrPrefix(str) {
  if (str && str.startsWith('enr:-')) {
    return true
  }
  return false
}

function isValidEnr(str) {
  try {
    if (!hasValidEnrPrefix(str)) {
      return false
    }
    // remove initial five characters
    str = str.substr(5, str.length);

    if (!isString(str)) {
      return false
    }
    if (!hasValidBase64Chars(str)) {
      return false
    }
    if (!isValidBase64Length(str)) {
      return false
    }

    return true
  } catch (e) {
    console.error(e)

    return false;
  }
}

const enr = "enr:-IS4QFV_wTNknw7qiCGAbHf6LxB-xPQCktyrCEZX-b-7PikMOIKkBg-frHRBkfwhI3XaYo_T-HxBYmOOQGNwThkBBHYDgmlkgnY0gmlwhKRc9_OJc2VjcDI1NmsxoQKHPt5CQ0D66ueTtSUqwGjfhscU_LiwS28QvJ0GgJFd-YN1ZHCCE4k"

let out = isValidEnr(enr)
console.log(out)
```

Reply:
So a couple of things:
1) The `enr:` is the 4 character prefix that indicates its a valid ENR.  I didn't realize that at first but reread the ENR spec again they're encoded using Base64URL, not raw Base64 and `-` is a valid character in the Base64URL encoding scheme.  As such, you need to adjust your code to check for valid [Base64url](https://base64.guru/standards/base64url) characters.  
2) The more I think about it, I'm not sure we gain anything by doing all this checking in the top level of our client.  The [ENR][(https://github.com/ChainSafe/discv5/blob/master/src/enr/enr.ts](https://github.com/ChainSafe/discv5/blob/ad1f536480cabf8b0d80916d626c373160b61cc8/src/enr/enr.ts#L165C54-L165C54)) class already checks for the correct prefix before deserializing so I think we're just duplicating checks here.  So, let's redirect this PR and drop all of the ENR validation and instead just merge any supplied ENRs from the two CLI parameters and then maybe wrap the entire loop in a `try/catch` that throws a coherent error regarding invalid ENRs if somebody supplies one.  So, you would just check the error that gets thrown out of the try/catch block,  and then throw a new Error caught by the outer execution loop that explains what the issue is (e.g. `invalid ENR ${badEnr} provided`) or whatever it is.